File: Back/WebShop-Backend/src/main/java/com/example/WebShop/models/dto/Category.java
package com.example.WebShop.models.dto;

import jakarta.persistence.Basic;
import jakarta.persistence.Column;
import lombok.Data;

import java.util.List;

@Data
public class Category {
    private Integer id;
    private String name;
    private List<Category> subcategories;
    private List<Attribute> attributes;
}

File: Front/WebShop-Frontend/src/app/models/category.ts
import { Attribute } from "@angular/core";

export class Category {
    id: number;
    name: string;
    subcategories: Category[];
    attributes: Attribute[];

    constructor(id:number, name:string, subcategories:Category[], attributes:Attribute[]){
        this.id = id;
        this.name = name;
        this.subcategories = subcategories;
        this.attributes = attributes;
    }
}
